rclpy/src/rclpy/_rclpy.c

```diff
@@ -4523,21 +4523,28 @@ rclpy_get_node_parameters(PyObject * Py_UNUSED(self), PyObject * args)
     return NULL;
   }
 
-  if (!PyDict_Contains(params_by_node_name, py_node_name_with_namespace)) {
-    // No parameters for current node.
-    Py_DECREF(params_by_node_name);
-    Py_DECREF(py_node_name_with_namespace);
-    return PyDict_New();
-  }
-  PyObject * node_params = PyDict_GetItem(params_by_node_name, py_node_name_with_namespace);
-  Py_DECREF(py_node_name_with_namespace);
-  if (NULL == node_params) {
-    Py_DECREF(params_by_node_name);
-    return NULL;
+  PyObject * node_params = PyDict_New();
+  PyObject * py_wildcard_name = PyUnicode_FromString("/**");
+
+  PyObject * current_key, * current_value;
+  Py_ssize_t current_index = 0;
+  while (PyDict_Next(params_by_node_name, &current_index, &current_key, &current_value)) {
+    // TODO(cottsay) implement further wildcard matching
+    if (PyObject_RichCompare(current_key, py_wildcard_name, Py_EQ) ||
```

Comment:
Memory leak? Each call to [`PyObject_RichCompare`](https://docs.python.org/3/c-api/object.html#c.PyObject_RichCompare) returns a new reference that needs to be decref'd.

Reply:
Switched to [`PyObject_RichCompareBool`](https://docs.python.org/3/c-api/object.html#c.PyObject_RichCompareBool).